include/openthread/platform/srp_replication.h

```diff
@@ -0,0 +1,166 @@
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes the platform abstraction for SRP Replication Protocol (SRPL).
+ */
+
+#ifndef OPENTHREAD_PLATFORM_SRP_REPLICATION_H_
+#define OPENTHREAD_PLATFORM_SRP_REPLICATION_H_
+
+#include <stdint.h>
+
+#include <openthread/error.h>
+#include <openthread/instance.h>
+#include <openthread/ip6.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * This structure represents an SRPL partner info discovered using DNS-SD browse on the service name "_srpl-tls._tcp".
+ *
+ */
+typedef struct otPlatSrplPartnerInfo
+{
+    /**
+     * This boolean flag indicates whether the entry is being removed or added.
+     *
+     * - TRUE indicates that peer is removed.
+     * - FALSE indicates that it is a new entry or an update to an existing entry.
+     *
+     */
+    bool mRemoved;
+
+    /**
+     * The TXT record data (encoded as specified by DNS-SD) from the SRV record of the discovered service.
+     *
+     */
+    const uint8_t *mTxtData;
+
+    uint16_t mTxtLength; ///< Number of bytes in @p mTxtData buffer.
+
+    /**
+     * The partner socket address (IPv6 address and port number).
+     *
+     * The port number is determined from the SRV record of the discovered SRPL service instance. The IPv6 address
+     * is determined from the DNS-SD query for A/AAAA records on the hostname indicated in the SRV record of the
+     * discovered service instance. If multiple host IPv6 addressees are discovered, one with highest scope is used.
+     *
+     */
+    otSockAddr mSockAddr;
+} otPlatSrplPartnerInfo;
+
+/**
+ * This function starts or stops DNS-SD browse to discover SRPL partners within the local domain.
+ *
+ * On start platform layer MUST initiate an ongoing DNS-SD browse on the service name "_srpl-tls._tcp" within the local
+ * browsing domain to discover SRPL partners. The ongoing browse will produce two different types of events:
+ * "add" events and "remove" events.  When the browse is started, it should produce an "add" event for every SRPL
+ * partner currently present on the network.  Whenever a partner goes offline, a "remove" event should be produced.
+ * remove"  events are not guaranteed, however.
+ *
+ * When an SRP partner is discovered, a new ongoing DNS-SD query for A/AAAA record MUST be started on the hostname
+ * indicated in the SRV record of the discovered entry. If multiple host IPv6 addressees are discovered for a partner,
+ * one with highest scope among all addresses MUST be reported (if there are multiple address at same scope, one must
+ * be selected randomly).
```

Comment:
> If multiple host IPv6 addressees are discovered for a partner, one with highest scope among all addresses MUST be reported (if there are multiple address at same scope, one must be selected randomly).

I wonder what consideration is behind this? I think link-local addresses are more likely to stay unchanged than addresses with higher scopes.

Reply:
Actually we may want change this and require all addresses to be reported. 
Each address can be reported seperately with its own `otPlatSrplHandleDnssdBrowseResult()` callback.